packages/gafl-webapp-service/src/pages/multibuy/remove-licence/__tests__/update-transaction.spec.js

```diff
@@ -36,13 +37,71 @@ const getPermissionThree = () => ({
   permit: { cost: 12 }
 })
 
-const createRequestMock = (transaction, permission, set = jest.fn()) => ({
+const getPagePermissionOne = () => ({
+  [ADD_LICENCE.page]: [Object]
+})
+
+const getPagePermissionTwo = () => ({
+  [ADD_LICENCE.page]: [Object],
+  [REMOVE_LICENCE]: [Object]
+})
+
+const getPagePermissionThree = () => ({
+  [ADD_LICENCE.page]: [Object]
+})
+
+const getStatusPermissionOne = () => ({
+  [ADD_LICENCE.page]: true
+})
+
+const getStatusPermissionTwo = () => ({
+  [ADD_LICENCE.page]: false,
+  [REMOVE_LICENCE.page]: true
+})
+
+const getStatusPermissionThree = () => ({
+  [ADD_LICENCE.page]: true
+})
+
+const getAddressLookupPermissionOne = () => ({})
+
+const getAddressLookupPermissionTwo = () => ({})
+
+const getAddressLookupPermissionThree = () => ({})
+
+const createRequestMock = (
+  transaction,
+  transactionPermission,
+  setTransaction = jest.fn(),
+  page,
+  setPage = jest.fn(),
+  status,
+  setStatus = jest.fn(),
+  setCurrentStatusPermission = jest.fn(),
+  addressLookup,
+  addressLookupPermission,
+  setAddressLookup = jest.fn()
```

Comment:
You shouldn't create a function with so many parameters - once you get above three or four, it's far preferable to have a 'config object' so that if a calling function only has to provide a single parameter, but it's in the 11th position, you don't have to provide the previous ten parameters.

Also, do you _really_ need to make _all_ of these configurable? As far as I can see, for `transactionPermission`, you always send the result of `getTransactionPermissionThree`. So get rid of `getTransactionPermissionThree`, get rid of the parameter and just use that object where you've got `transactionPermission`